### DISCUSSION TITLE
Additional security options?

### DISCUSSION THREAD

**Discussion by eero-t** [any1/wayvnc#76] [Ideas]
This lists few (hopefully easy) security options which could be useful with clients that don't support encryption [1]:
* https://help.ubuntu.com/community/VNC/Servers

Namely:
* once: allow only first connection
* local: allow only local (e.g. ssh port-forwarded) connections

[1] I didn't find any security options support in Gnome Boxes, not even username/password?

**Comment by any1** [comment]
Only allowing local connections is the default. "once" mode is not a feature.

**Comment by eero-t** [comment]
WayVNC manual page could list all the defaults (e.g. as another config example).

What means "once" mode not being a feature?  At least currently I can open multiple sessions to WayVNC with Boxes.

**Comment by any1** [comment]
> What means "once" mode not being a feature? At least currently I can open multiple sessions to WayVNC with Boxes.

I mean that there is no option to specify that wayvnc should exit after the first connection has been terminated.

**Comment by any1** [comment]
It might be a good idea to hand over authentication to an external command. I.e. whenever a new user connects, wayvnc spawns a user defined process, pipes info into it such as source network address, username and password and then reads the exit code of the process. That way, users can implement custom authentication via shell scripts.
